MusicTaste/MusicTaste/MatchingAlgorithm.m

```diff
@@ -16,38 +16,34 @@ + (instancetype)shared {
     return sharedManager;
 }
 
-+ (void) lookForMatches{
+- (void) lookForMatches{
     NSMutableDictionary *matchDict = [NSMutableDictionary new];
     PFObject *music = [PFObject objectWithClassName:@"Music"];
     PFQuery *query = [PFQuery queryWithClassName:@"Music"];
     PFUser *current = [PFUser currentUser];
     //Goes through Query
     [query findObjectsInBackgroundWithBlock:^(NSArray *spotifyData, NSError *error) {
         for (PFObject *user in spotifyData){
-            //We make sure we are not comparing the same object
-            if (![user.objectId isEqual:music.objectId]){
-                //Calls the compare function to compare each user
-                [MatchingAlgorithm compare:user withDictionary:matchDict withData:user];
-            }
+            //Calls the compare function to compare each user
+            [self compareSpotifyData:user withData:spotifyData];
         }
+        //sets the matches as an integer in the user's database
+        [current saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {}];
     }];
-    //sets the matches as an integer in the user's database
-    NSArray *matches = [[[matchDict keysSortedByValueUsingSelector:@selector(compare:)] reverseObjectEnumerator] allObjects];
-    music[@"matches"] = [NSArray arrayWithArray:matches];
-    [current saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {}];
+    
 }
 
-+ (double) compareSpotifyData:(PFObject *)potentialMatch withData:(NSManagedObject *)userSpotify{
-    double spotifyMatch = 0.0;
+- (void) compareSpotifyData:(PFObject *)potentialMatch withData:(NSManagedObject*)userSpotify{
     PFQuery *query = [PFQuery queryWithClassName:@"Music"];
     PFUser *current = [PFUser currentUser];
-    [query whereKey:@"userId" equalTo:current.objectId];
     //Makes sure the we are collecting information from the current user
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0), ^{
-        [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
+    self.spotifyMatch = 0.0;
+    NSMutableArray *matches = [NSMutableArray array];
             double spotifyMatch = 0.0;
-            NSManagedObject *matchData = results[0];
-            if (potentialMatch != NULL){
+    for (NSManagedObject *matchData in userSpotify){
+        if (potentialMatch != NULL){
+            if (![potentialMatch.objectId isEqual:[matchData valueForKey:@"objectId"]]){
```

Comment:
I check if potential match is null and if we are comparing the potential match to itself